src/library/export/engineprimeexportjob.cpp

```diff
@@ -0,0 +1,495 @@
+        auto srcPath = srcFileInfo.location();
+        QFile::copy(srcPath, dstPath);
+    }
+
+    return request.engineLibraryDbDir.relativeFilePath(dstPath).toStdString();
+}
+
+djinterop::track getTrackByRelativePath(
+        djinterop::database database, const std::string& relativePath) {
+    auto trackCandidates = database.tracks_by_relative_path(relativePath);
+    switch (trackCandidates.size()) {
+    case 0:
+        return database.create_track(relativePath);
+    case 1:
+        return trackCandidates.front();
+    default:
+        qInfo() << "Warning: More than one external track with the same relative path";
+        return trackCandidates.front();
+    }
+}
+
+void exportMetadata(djinterop::database& db,
+        QHash<TrackId, int64_t>& mixxxToEnginePrimeTrackIdMap,
+        TrackPointer pTrack,
+        std::unique_ptr<Waveform> pWaveform,
+        const std::string& relativePath) {
+    // Create or load the track in the database, using the relative path to
+    // the music file.  We will record the mapping from Mixxx track id to
+    // exported track id as well.
+    auto externalTrack = getTrackByRelativePath(db, relativePath);
+    mixxxToEnginePrimeTrackIdMap.insert(pTrack->getId(), externalTrack.id());
+
+    // Note that the Engine Prime format has the scope for recording meta-data
+    // about whether track was imported from an external database.  However,
+    // that meta-data only extends as far as other Engine Prime databases,
+    // which Mixxx is not.  So we do not set any import information on the
+    // exported track.
+    externalTrack.set_track_number(pTrack->getTrackNumber().toInt());
+    externalTrack.set_bpm(pTrack->getBpm());
+    externalTrack.set_year(pTrack->getYear().toInt());
+    externalTrack.set_title(pTrack->getTitle().toStdString());
+    externalTrack.set_artist(pTrack->getArtist().toStdString());
+    externalTrack.set_album(pTrack->getAlbum().toStdString());
+    externalTrack.set_genre(pTrack->getGenre().toStdString());
+    externalTrack.set_comment(pTrack->getComment().toStdString());
+    externalTrack.set_composer(pTrack->getComposer().toStdString());
+    externalTrack.set_key(toDjinteropKey(pTrack->getKey()));
+    int64_t lastModifiedMillisSinceEpoch =
+            pTrack->getFileInfo().fileLastModified().toMSecsSinceEpoch();
+    std::chrono::system_clock::time_point lastModifiedAt{
+            std::chrono::milliseconds{lastModifiedMillisSinceEpoch}};
+    externalTrack.set_last_modified_at(lastModifiedAt);
+    externalTrack.set_bitrate(pTrack->getBitrate());
+
+    // Frames used interchangeably with "samples" here.
+    auto sampleCount = static_cast<int64_t>(pTrack->getDuration() * pTrack->getSampleRate());
+    externalTrack.set_sampling({static_cast<double>(pTrack->getSampleRate()), sampleCount});
+
+    // TODO(mr-smidge) - Set average loudness.  Where does Mixxx store this?
+    externalTrack.set_average_loudness(0.5f);
```

Comment:
```
auto replayGain = pTrack->getReplayGain();
// Assumption: Gain has been calculated with the new EBU R128 algorithm.
const double referenceGainDb = ratio2db(replayGain.getRatio());
// Reconstruct the LUFS value from the relative gain
const double ituBs1770Lufs = AnalyzerEbur128::kReplayGain2ReferenceLUFS - referenceGainDb;
```

Reply:
Thanks, Uwe - will look into this.

Reply:
I intend to perform some detailed analysis in future of exactly how the loudness field behaves on a set of reference tracks, but for now the replay gain ratio itself seems like a reasonable initial setting.

Good to know we have EBU R128 in Mixxx.